### PR TITLE
Battle/int class

### PR DESCRIPTION
**Pull request by Purrsephone**
Hey fellow devs, 

I integrated the real class_t struct into our battle code with a few minor changes. 

Thanks,

Sophie.

### CODE REVIEW

File: tests/battle/test_battle_flow.c [hunk 1]
```diff
@@ -25,12 +26,13 @@ Test(battle_flow, set_player)
 /* Tests set_enemies() with 1 enemy */
 Test(battle_flow, set_one_enemy)
 {
-    npc_enemy_t *npc_enemy = make_npc_enemy("set_one_enemy_Name", NULL, NULL, NULL);
-
+    npc_enemy_t *npc_enemy = make_npc_enemy("enemy_name", NULL, NULL, NULL);
     combatant_t *comb_enemy = set_enemies(npc_enemy);
 
+    printf("%s\n", comb_enemy->name);
+
```
Comment (alex-sheen) [inline tests/battle/test_battle_flow.c:33]: Delete printf statement

File: tests/battle/test_battle_flow.c [hunk 2]
```diff
@@ -25,12 +26,13 @@ Test(battle_flow, set_player)
 /* Tests set_enemies() with 1 enemy */
 Test(battle_flow, set_one_enemy)
 {
-    npc_enemy_t *npc_enemy = make_npc_enemy("set_one_enemy_Name", NULL, NULL, NULL);
-
+    npc_enemy_t *npc_enemy = make_npc_enemy("enemy_name", NULL, NULL, NULL);
```
Comment (alex-sheen) [inline tests/battle/test_battle_flow.c:29]: Is there a reason why these test names were changed?

File: include/playerclass/class_structs.h [hunk 3]
```diff
@@ -61,4 +61,4 @@ typedef struct class {
 
 } class_t;
 
-#endif /* CLASS_STRUCTS_H */
\ No newline at end of file
+#endif /* CLASS_STRUCTS_H */
```
Comment (alex-sheen) [inline include/playerclass/class_structs.h]: I think this should be removed for clarity, since its just an accidental deletion of a new line but its in the player class files.

File: tests/battle/test_battle_flow.c (hunk 2 above)
Comment (Purrsephone) [reply tests/battle/test_battle_flow.c:29]: Yes, player class defines the max name length as 20 rather than 50 chars.

File: include/playerclass/class_structs.h (hunk 3 above)
Comment (Purrsephone) [reply include/playerclass/class_structs.h]: Yeah, I can bring this up when I meet with player class, but I'll leave it alone for now. Good catch!

File: include/battle/battle_moves.h [hunk 4]
```diff
@@ -3,9 +3,11 @@
 
 #include <stdbool.h>
 #include <string.h>
-#include "battle_classes.h"
+#include "playerclass/class_structs.h"
 #include "battle_structs.h"
 
+/* defines max length of class info */
+static const int MAX_CLASS_INFO_LEN = 100;
```
Comment (alex-sheen) [inline include/battle/battle_moves.h:10]: What is this used for? class_structs.h has short and long class descriptions defined, should we use those?

Comment (Purrsephone) [reply include/battle/battle_moves.h:10]: Oh you're right, I renamed it MAX_MOVE_INFO_LEN
It's for the description of a move, so I'm going to keep it separate from class, but gave it a better name.

File: tests/battle/test_battle_flow.c (hunk 2 above)
Comment (alex-sheen) [reply tests/battle/test_battle_flow.c]: gotcha

File: include/battle/battle_flow_structs.h [hunk 5]
```diff
@@ -40,42 +41,43 @@ typedef struct chiventure_ctx_battle {
...
+ *     - p_id: player id
  *     - stats: player stats stub
  *     - moves: player moves stub
  *     - items: player items stub
- * 
+ *
  * Returns: a newly allocated player_t with p_id, stats, moves, items
  */
-player_t *new_ctx_player(char* p_id, stat_t *stats, move_t *moves, item_t* items);
+player_t *new_ctx_player(char* p_id, class_t *class, stat_t *stats, move_t *moves, item_t* items);
```
Comment (dwahme) [inline include/battle/battle_flow_structs.h:62]: You need to update the function parameters documentation

File: include/battle/battle_flow_structs.h [hunk 6]
```diff
@@ -40,42 +41,43 @@ typedef struct chiventure_ctx_battle {
...
+ *     - npc_id: npc id
  *     - stats: npc stats stub
  *     - moves: npc moves stub
  *     - items: npc items stub
- * 
+ *
  * Returns: a single newly allocated npc_enemy_t with npc_id, stats, moves, items
  */
-npc_enemy_t *make_npc_enemy(char* npc_id, stat_t *stats, move_t *moves, item_t* items);
+npc_enemy_t *make_npc_enemy(char* npc_id, class_t *class, stat_t *stats, move_t *moves, item_t* items);
```
Comment (dwahme) [inline include/battle/battle_flow_structs.h:74]: You need to update the function parameters documentation

File: include/battle/battle_moves.h [hunk 7]
```diff
@@ -3,9 +3,11 @@
 
 #include <stdbool.h>
 #include <string.h>
-#include "battle_classes.h"
+#include "playerclass/class_structs.h"
 #include "battle_structs.h"
 
+/* defines max length of class info */
+static const int MAX_MOVE_INFO_LEN = 100;
```
Comment (dwahme) [inline include/battle/battle_moves.h:10]: Use `#define MAX_MOVE_INFO_LEN  (100)` instead (that's the convention chiventure uses)

File: src/battle/src/battle_state.c [hunk 8]
```diff
@@ -80,9 +85,9 @@ int combatant_free(combatant_t *c)
         free(item_elt);
     }
 
-	free(c);
+        free(c);
 
-	return SUCCESS;
+        return SUCCESS;
```
Comment (dwahme) [inline src/battle/src/battle_state.c:90]: I think these are indented too far

File: tests/battle/test_battle_flow.c [hunk 9]
```diff
@@ -1,6 +1,7 @@
+#include <stdlib.h>
```
Comment (dwahme) [inline tests/battle/test_battle_flow.c:1]: We need at least one test that actually uses playerclasses instead of just NULL

File: tests/battle/test_battle_flow_structs.c [hunk 10]
```diff
@@ -7,7 +7,7 @@
 /* Tests new_ctx_player() */
```
Comment (dwahme) [inline tests/battle/test_battle_flow_structs.c:7]: We need at least one test that actually uses playerclasses instead of just NULL

File: tests/battle/test_battle_logic.c [hunk 11]
```diff
@@ -13,15 +13,15 @@
 Test(battle_logic, target_exists)
```
Comment (dwahme) [inline tests/battle/test_battle_logic.c:13]: We need at least one test that actually uses playerclasses instead of just NULL

File: tests/battle/test_battle_state.c [hunk 12]
```diff
@@ -10,7 +10,7 @@ Test(battle_state, combatant_new)
 {
```
Comment (dwahme) [inline tests/battle/test_battle_state.c:10]: We need at least one test that actually uses playerclasses instead of just NULL

**Comment by dwahme** [comment]
Issue Score: ✔️

Comments:
Great job! Here are a few things you may want to improve in future issues:

The title of an issue has to convey what the issue will be producing. In the future, try using simple titles with a "Verb-Object" structure that clearly states what you will be doing in the issue (e.g., “Design and implement a ‘circle’ module”, “Research libraries for drawing simple geometric shapes”, etc.)

The summary for this issue is too terse. A good rule of thumb is that you should be writing at least one full paragraph in your summary (see https://uchicago-cs.github.io/cmsc22000/project/policies/#a-descriptive-summary for examples). If this issue was generated from a Scrum board card, please remember that the summary you write in the Sprint meeting is meant to be a quick draft; it needs to be fleshed out when you convert the card to an issue.

When an issue is done, you should include a final comment summarizing what was done, and why the issue is being closed.
